src/Components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,55 @@
+import { useContext } from 'react';
+import { DispatchContext, StateContext } from '../Store';
+import { TodoItem } from '../TodoItem';
+import { Status, Todo } from '../../types';
+
+export const TodoList: React.FC = () => {
+  const { todoList, filter } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+  let filteredTodoList: Todo[];
+
+  switch (filter) {
+    case Status.active:
+      filteredTodoList = todoList.filter(todo => todo.completed === false);
+      break;
+
+    case Status.completed:
+      filteredTodoList = todoList.filter(todo => todo.completed === true);
+      break;
+
+    case Status.all:
+      filteredTodoList = [...todoList];
+      break;
+
+    default:
+      filteredTodoList = [...todoList];
+  }
+
+  const completedAll = todoList.length > 0
```

Comment:
Instead of checking the length of the array and then using every, you could use every directly. If the array is empty, every will return true by default, which is the opposite of what you want. You could use some to check if there is at least one uncompleted item, and negate the result.